### ISSUE TITLE
Unable to mount filesystem

### ISSUE THREAD

**Issue by gray-adeyi** [GSConnect/gnome-shell-extension-gsconnect#1294]
Hi devs! Great works you've been doing. The issue I've been facing with GSConnect is the inability to mount filesystem.
My suspicision is that this occurred when i tried to extend my phone storage with an sd card.

Here's the log i got while try to replicate the error. I hope it helps.

GSConnect: 48 (system)
GJS:       17001
Session:   wayland
OS:        Manjaro Linux
--------------------------------------------------------------------------------
Mar 15 02:40:01 systemd[1]: NetworkManager-dispatcher.service: Deactivated successfully.
Mar 15 02:40:01 audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 15 02:40:01 kernel: audit: type=1131 audit(1647308401.870:1591): pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 15 02:40:48 org.gnome.Shell.Extensions.GSConnect[1829]: [/service/device.js:sendPacket:446]: Nokia 3.1: {
                                                              "id": 1647308448103,
                                                              "type": "kdeconnect.sftp.request",
                                                              "body": {
                                                                "startBrowsing": true
                                                              }
                                                            }
Mar 15 02:40:48 org.gnome.Shell.Extensions.GSConnect[1829]: [/service/device.js:_readLoop:338]: Nokia 3.1: {
                                                              "id": 1647308446582,
                                                              "type": "kdeconnect.sftp",
                                                              "body": {
                                                                "ip": "192.168.43.203",
                                                                "port": 1744,
                                                                "user": "kdeconnect",
                                                                "password": "4ZX163CEkYcvCtstRc1bqX15Q9xd",
                                                                "path": "/",
                                                                "multiPaths": [
                                                                  "/downloads"
                                                                ],
                                                                "pathNames": [
                                                                  "downloads"
                                                                ]
                                                              }
                                                            }
Mar 15 02:40:48 gjs[1829]: JS ERROR: Nokia 3.1: Gio.IOErrorEnum: Connection failed
                           _handleMount/</<@/usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/sftp.js:253:38
                           @/usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:727:17

**Comment by ferdnyc** [comment]
@gray-adeyi 

Hmm, from all appearance your Nokia device is just refusing the SFTP connection we're attempting.

You say you installed extended storage, did you go into the KDE Connect app settings and Allow it access to that new storage? You may need to do that, or if the new storage is mounted inside the path you previously authorized, you may even need to re-authorize the _old_ path so that KDE Connect is still permitted to access the necessary device paths.

The problem seems to point to a device-side issue, which usually means permissions. That's the first thing I'd recommend checking; there's not a lot to go on from the GSConnect side, which is unfortunately common with SFTP failures.

**Comment by gray-adeyi** [comment]
Thanks for writing back. Good to know the fault is not from GSConnect 
but my phone. I'm not sure it's a permission issue. I've allowed all 
permissions to my phone's storage. The external storage I added to my 
phone is not working as an external storage but as an extension of the 
phone's internal storage...I don't know how best to explain. The 
filesystem mount used to work before the extension

**Comment by alexanderadam** [comment]
I'm having the same issue since I upgraded to Ubuntu 22.04.

There's no `` any more but I installed [`python3-nautilus`](https://www.ubuntuupdates.org/package/core/jammy/universe/base/python3-nautilus) instead. It doesn't seem to work though. The package `gir1.2-nautilus-3.0` is still available and working.

[Folks commenting on the official extension page](https://extensions.gnome.org/extension/1319/gsconnect/#comments) also seem to struggle.

I tried debugging with
```bash
pkill gvfs; pkill nautilus
GVFS_SMB_DEBUG=10 GVFS_DEBUG=1 $(find /usr/lib* -name gvfsd 2>/dev/null) --replace 2>&1 | tee gvfsd.log
```

and the output is

```
sftp: Added new job source 0x15329f3940a0 (GVfsBackendSftp)
sftp: Queued new job 0x55339f172f30 (GVfsJobMount)
sftp: spawn_ssh: /usr/bin/ssh -oForwardX11 no -oForwardAgent no -oPermitLocalCommand no -oClearAllForwardings yes -oProtocol 2 -oNoHostAuthenticationForLocalhost yes -oControlMaster auto -oControlPath=/run/user/1000/gvfsd-sftp/%C -p 1740 -s 123.123.123.123 sftp 
sftp: handle_login #1 initial_connection = 1 - user: (null), host: 123.123.123.123, port: 1740
sftp: handle_login #1 - password_save: 0
sftp: handle_login #1 - ret_val: 1
sftp: stderr: Unable to negotiate with 123.123.123.123 port 1740: no matching host key type found. Their offer: ssh-rsa
sftp: send_reply(0x55339f172f30), failed=1 (Connection failed)
```

This error message lead me to StackOverflow where it was suggested to add this to your `~/ssh/config`:

```ssh
HostKeyAlgorithms ssh-rsa
PubkeyAcceptedKeyTypes ssh-rsa 
```

But I guess it would make sense to add this to the command line parameter. Doesn't it?

**PS:** Maybe it would help debugging if the settings would show a tailed debug output or so since mount issues return [again](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/478) and [again](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1115). Or is the mount command returning success / error code? Maybe showing an error modal with a log excerpt would be good then.

**Comment by dr-zernov** [comment]
Run ssh-keygen and add the above mentioned lines to ~/.ssh/config

Device automatically disconnects and reconnects mounting the storage on nautilus.

I am on Fedora 36

**Comment by EntityinArray** [comment]
> I was also having yet another mount issue since I upgraded to Ubuntu 22.04.
> 
> There's no `python-nautilus` any more but I installed [`python3-nautilus`](https://www.ubuntuupdates.org/package/core/jammy/universe/base/python3-nautilus) instead. It didn't help though. The package `gir1.2-nautilus-3.0` is still available and working.
> 
> [Folks commenting on the official extension page](https://extensions.gnome.org/extension/1319/gsconnect/#comments) also seem to struggle.
> 
> I tried debugging with
> 
> ```shell
> pkill gvfs; pkill nautilus
> GVFS_SMB_DEBUG=10 GVFS_DEBUG=1 $(find /usr/lib* -name gvfsd 2>/dev/null) --replace 2>&1 | tee gvfsd.log
> ```
> 
> when mounting and the output was
> 
> ```
> sftp: Added new job source 0x15329f3940a0 (GVfsBackendSftp)
> sftp: Queued new job 0x55339f172f30 (GVfsJobMount)
> sftp: spawn_ssh: /usr/bin/ssh -oForwardX11 no -oForwardAgent no -oPermitLocalCommand no -oClearAllForwardings yes -oProtocol 2 -oNoHostAuthenticationForLocalhost yes -oControlMaster auto -oControlPath=/run/user/1000/gvfsd-sftp/%C -p 1740 -s 123.123.123.123 sftp 
> sftp: handle_login #1 initial_connection = 1 - user: (null), host: 123.123.123.123, port: 1740
> sftp: handle_login #1 - password_save: 0
> sftp: handle_login #1 - ret_val: 1
> sftp: stderr: Unable to negotiate with 123.123.123.123 port 1740: no matching host key type found. Their offer: ssh-rsa
> sftp: send_reply(0x55339f172f30), failed=1 (Connection failed)
> ```
> 
> The error message `no matching host key type found. Their offer: ssh-rsa` lead me to StackOverflow where it was suggested to add the following lines to your `~/ssh/config`:
> 
> ```
> HostKeyAlgorithms ssh-rsa
> PubkeyAcceptedKeyTypes ssh-rsa 
> ```
> 
> But I guess it would make sense to add this to the command line parameter or as an explicit option where it is currently called. Doesn't it?
> 
> **PS:** Maybe it would help debugging if the settings would show a tailed debug output or so since mount issues return [again](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/478) and [again](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1115). Or is the mount command also returning success / error code? Because showing an error modal with a log excerpt would be good then as well.

Yes, i'm currently experiencing the same issue on Fedora 36.

**Comment by schklom** [comment]
@alexanderadam FYI, I am on the same OS as you and
```
HostKeyAlgorithms ssh-rsa
```
is enough to mount my phone's filesystem.

Thank you very much for finding this hacky solution :)

**Comment by omer-musa-battal** [comment]
the solution suggested by @joshcangit worked for me. I added the following line in my `${HOME}/.ssh/config` file:
```
HostKeyAlgorithms +ssh-rsa
```
However, I had another issue. The permissions of my .ssh folder and its contents were set incorrectly. I found out about this when I tried to connect to a regular ssh session, and failed due to various permission errors.

[This answer](https://superuser.com/a/215506) provides general info regarding correct .ssh folder permissions, quoted below for ease of reference.
> Typically you want the permissions to be:
> 
> .ssh directory: 700 (drwx------)
> public key (.pub file): 644 (-rw-r--r--)
> private key (id_rsa): 600 (-rw-------)

After I corrected the permissions of my .ssh folder, gsconnect mount problem was also resolved. 

[This comment](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/816#issuecomment-613817778) also helped me to find out what is going on.
